### ISSUE TITLE
Elisp checker is confused by threading macros (->, ->>, etc.)

### ISSUE THREAD

**Issue by rogual** [flycheck/flycheck#1234]
Given this code:
```
(defun test ()
  (->> "abcabc"
       (replace-regexp-in-string "b" "c")
       (replace-regexp-in-string "cc" "d")))
```

Flycheck flags the calls to `replace-regexp-in-string` as warnings:

`replace-regex-in-string called with 2 arguments, but requires 3-7`

Expected behaviour: No warnings. Forms given to ->> aren't ordinary function call forms.

Flycheck version: 31snapshot (package: 20170223.247)
Emacs version: Gnu Emacs25.1.1
OS: Linux (CentOS 7, Gnome 3)

**Update**: Looks like these macros don't come with Emacs, they're from a package called "dash". Should I still expect Flycheck to work with them?

**Comment by cpitclaudel** [comment]
Thanks for your message.  The problem is not in Flycheck, but in Emacs: your file doesn't specify where to get these macros, so as far as Emacs is concerned they are undefined, and naturally you get the warnings you mentioned.  If you `(require 'dash)` first, everything should be fine.

**Comment by rogual** [comment]
Works great. Sorry for the noise.

**Comment by cpitclaudel** [comment]
No problem :) Enjoy Flycheck!
